databroker/src/grpc/kuksa_val_v2/conversions.rs

```diff
@@ -207,6 +207,99 @@ impl From<&proto::Datapoint> for broker::DataValue {
     }
 }
 
+impl From<&broker::Metadata> for proto::Metadata {
+    fn from(metadata: &broker::Metadata) -> Self {
+        proto::Metadata {
+            id: metadata.id,
+            data_type: proto::DataType::from(metadata.data_type.clone()) as i32,
+            entry_type: proto::EntryType::from(metadata.entry_type.clone()) as i32,
+            description: Some(metadata.description.clone()),
+            comment: None,
+            deprecation: None,
+            unit: metadata.unit.clone(),
+            value_restriction: match metadata.allowed.as_ref() {
+                Some(allowed) => match allowed {
+                    broker::DataValue::StringArray(vec) => Some(proto::ValueRestriction {
+                        r#type: Some(proto::value_restriction::Type::String(
```

Comment:
As a Rust newbie - why do we need `r#type` here?

Reply:
This is because of protobuf generator to avoid conflicts. It generates `r#type` if a field's name is declared as `type`.

Reply:
Ah, `type` is possibly a reserved word in Rust?